lib/internal/fs/streams.js

```diff
@@ -309,7 +311,7 @@ WriteStream.prototype._write = function(data, encoding, cb) {
     });
   }
 
-  fs.write(this.fd, data, 0, data.length, this.pos, (er, bytes) => {
+  this.fs.write(this.fd, data, 0, data.length, this.pos, (er, bytes) => {
```

Comment:
If `this.fs` is set, we should not be using `writev()`.

Reply:
I don't understand?

Reply:
@ronag fs write streams support both `._write()` and `._writev()`, but only `._write()` uses public fs APIs. Because this is about being able to override the fs API being used, `._writev()` should not be used.

Reply:
Where is writev used?

Reply:
It’s the next 2 functions after this one.

One straightforward solution is to put `if (this.fs !== fs) this._writev = undefined` in the constructor.